src/Security/MemberAuthenticator/LostPasswordForm.php

```diff
@@ -23,8 +25,15 @@ class LostPasswordForm extends MemberLoginForm
      */
     public function getFormFields()
     {
+        $uniqueIdentifier = Member::config()->get('unique_identifier_field');
+        if ($uniqueIdentifier === 'Email') {
+            $emailField = EmailField::create('Email', _t('SilverStripe\\Security\\Member.EMAIL', 'Email'));
+        } else {
+            // This field needs to still be called Email, but we can re-label it
+            $emailField = TextField::create('Email', _t('SilverStripe\\Security\\Member.LOSTPASSWORDLABEL', $uniqueIdentifier));
+        }
```

Comment:
Might be worth getting a member singleton and calling `$singleton->fieldLabel($uniqueIdentifier)` to get the field label. Then we don't need any branching logic at all.

Reply:
That's a good trick! Still need to split on fieldtype though I think, the email validation on the email field still seems valuable.